app/code/community/Nosto/Tagging/Block/Customer.php

```diff
@@ -115,6 +115,19 @@ public function getNostoCustomer()
             $nostoCustomer->setDateOfBirth(DateTime::createFromFormat("Y-m-d H:i:s", $dateOfBirth));
         }
         $nostoCustomer->setMarketingPermission($emailHelper->isOptedIn($email));
+        $customerAddress = $customer->getPrimaryShippingAddress();
+        if ($customerAddress) {
```

Comment:
Can we check the type here? 
```suggestion
        if ($customerAddress instanceof Mage_Customer_Model_Address) {
```